lib/streamlit/elements/map.py

```diff
@@ -220,28 +434,28 @@ def to_deckgl_json(data: Data, zoom: Optional[int]) -> str:
             longitude_distance = range_lat
         zoom = _get_zoom_level(longitude_distance)
 
-    # "+1" because itertuples includes the row index.
-    lon_col_index = data.columns.get_loc(lon) + 1
-    lat_col_index = data.columns.get_loc(lat) + 1
-    final_data = []
-    for row in data.itertuples():
-        final_data.append(
-            {"lon": float(row[lon_col_index]), "lat": float(row[lat_col_index])}
-        )
```

Comment:
In the past, we used to create columns called "lat" and "lon" with the values of the latitude and longitude data, and we'd then tell pydeck to use those columns by setting `getPosition: "@@=[lon, lat]"`.

In the new code, we instead keep the latitude and longitude columns untouched and instead set `getPosition: f"@@=[{lon_col_name}, {lat_col_name}]"`.